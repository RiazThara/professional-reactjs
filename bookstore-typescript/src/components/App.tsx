import { useState, useEffect } from 'react';
import Header from './Header';
import Main from './Main';
import './App.css';
import useFetch from '../hooks/useFetch';

function App() {
  const [itemsInCart, setItemsInCart] = useState<Array<string>>([]);
  const { products, isLoading } = useFetch('/data/products.json');

  useEffect(() => {
    shuffleArray(products);
  }, [products]);

  function shuffleArray(array: any[]) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  function addToCart(id: string) {
    let newItems = [...itemsInCart, id];
    setItemsInCart(newItems);
  }

  function removeFromCart(idToRemove: string) {
    let newItems = itemsInCart.filter((id) => id !== idToRemove);
    setItemsInCart(newItems);
  }

  return (
    <div className="container">
      <Header />
      {isLoading ? 'Loading' : ''}
      <Main
        products={products}
        itemsInCart={itemsInCart}
        addToCart={addToCart}
        removeFromCart={removeFromCart}
      />
    </div>
  );
}

export default App;
