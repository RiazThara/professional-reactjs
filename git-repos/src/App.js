import { useEffect, useState } from 'react';

import AddItemForm from './AddItemForm';
import TodoList from './TodoList';

function App() {
  const [todoItems, setTodoItems] = useState([]);
  const [error, setErrorMessage] = useState('');
  const [repos, setRepos] = useState([]);

  useEffect(() => {
    async function getData() {
      let response = await fetch(
        'https://api.github.com/users/chrisminnick/repos'
      );
      let data = await response.json();
      setRepos(data);
    }
    getData();
  }, []);

  function validateTodoItem(item) {
    setErrorMessage('');
    if (item.length < 2) {
      setErrorMessage('length must be more than two characters.');
      console.log(item);
      console.log(error);
    }
  }

  function addNewItem(newItem) {
    validateTodoItem(newItem);
    if (error) {
      return false;
    }
    setTodoItems([
      ...todoItems,
      { id: todoItems.length, description: newItem, completed: false },
    ]);
  }

  function toggleCompleteStatus(completedItem) {
    let todoItemsCopy = [...todoItems];
    todoItemsCopy = todoItems.map(function (item) {
      if (item.id === completedItem.id) {
        item.completed = !item.completed;
      }
      return item;
    });

    setTodoItems([...todoItemsCopy]);
  }

  return (
    <>
      {console.log(repos)}
      <AddItemForm addNewItem={addNewItem} />
      <p style={{ color: 'red' }}>{error}</p>

      <div>
        Todo:
        <TodoList
          todoItems={todoItems.filter((item) => !item.completed)}
          markAsComplete={toggleCompleteStatus}
        />
      </div>
      <br />
      <div>
        Completed:
        <TodoList
          todoItems={todoItems.filter((item) => item.completed)}
          markAsComplete={toggleCompleteStatus}
        />
      </div>
      <h1>Repo List</h1>

      {repos.map((repo) => (
        <li key={repo.id}>{repo.name}</li>
      ))}
    </>
  );
}

export default App;
