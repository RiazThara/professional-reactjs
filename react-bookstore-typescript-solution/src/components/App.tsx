import Footer from './Footer.tsx';
import Header from './Header.tsx';
import MainContainer from './MainContainer.tsx';
import { useState, useEffect } from 'react';
import { Book } from '../types.ts';

function App() {
  const [itemsInCart, setItemsInCart] = useState<string[]>([]);
  const [products, setProducts] = useState<Book[]>([]);
  const [loading, setLoading] = useState<boolean>(false);

  useEffect(() => {
    async function fetchData() {
      setLoading(true);

      try {
        const response = await fetch('//localhost:5173/data/products.json');
        const products = await response.json();
        shuffleArray(products);
        setProducts(products);
      } catch (error) {
        console.log(error);
      }
      setLoading(false);
    }
    fetchData();
  }, [setProducts]);

  function shuffleArray(array: Book[]) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  function addToCart(id: string) {
    setItemsInCart([...itemsInCart, id]);
  }

  function removeFromCart(id: string) {
    const newItemsInCart = itemsInCart.filter((item) => item !== id);
    setItemsInCart(newItemsInCart);
  }

  return (
    <div className="container">
      <Header />
      {loading && <p>Loading...</p>}
      <MainContainer
        products={products}
        itemsInCart={itemsInCart}
        addToCart={addToCart}
        removeFromCart={removeFromCart}
      />
      <Footer />
    </div>
  );
}

export default App;
