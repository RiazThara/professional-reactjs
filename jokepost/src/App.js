import { useState, useEffect } from 'react';
import JokeInput from './JokeInput';
import LatestJokes from './LatestJokes';

import './App.css';

function App() {
  const [jokes, setJokes] = useState([]);
  const [newJoke, setNewJoke] = useState('');
  const [count, setCount] = useState(1);

  async function fetchJokes() {
    const response = await fetch('/data.json');
    const data = await response.json();
    setJokes(data);
  }

  useEffect(() => {
    fetchJokes();
  }, [count]);

  return (
    <div className="App">
      <h1>JokePost {count}</h1>
      <button onClick={() => setCount(count + 1)}>
        Increment and refresh jokes
      </button>
      <br />
      <br />
      <JokeInput
        newJoke={newJoke}
        setNewJoke={setNewJoke}
        setJokes={setJokes}
        jokes={jokes}
      />
      <p>Here's the joke you're writing: {newJoke}</p>
      <LatestJokes jokes={jokes} />
    </div>
  );
}

export default App;
