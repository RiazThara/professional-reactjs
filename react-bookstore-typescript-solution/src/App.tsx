import Footer from './Footer.tsx';
import Header from './Header.tsx';
import MainContainer from './MainContainer.tsx';
import './App.css';
import { products } from './data/products.ts';

function App() {
  return (
    <div className="container">
      <Header />
      <MainContainer products={products} />
      <Footer />
    </div>
  );
}

export default App;
