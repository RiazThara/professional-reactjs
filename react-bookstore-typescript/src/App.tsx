import Footer from './Footer';
import Header from './Header';
import MainContainer from './MainContainer';
import './App.css';
import { products } from './data/products.ts';

function App() {
  return (
    <div className="container">
      <Header />
      <MainContainer products={products} />
      <Footer />
    </div>
  );
}

export default App;
