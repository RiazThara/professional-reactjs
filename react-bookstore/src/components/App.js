import { useState, useEffect } from 'react';
import Header from './Header';
import Main from './Main';
import Footer from './Footer';
import './App.css';
import useFetch from '../hooks/useFetch';
import * as actionCreators from '../actions';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

function App(props) {
  const [filterText, setFilterText] = useState('');
  const [filteredProductList, setFilteredProductList] = useState([]);
  const { data, isLoading } = useFetch('/data/products.json');
  const { loadProducts, products } = props;

  useEffect(() => {
    loadProducts(data);
    setFilteredProductList(data);
  }, [loadProducts, data]);

  useEffect(() => {
    filterProducts(filterText);
    function filterProducts(filterText) {
      let filteredProducts = products.filter((product) =>
        product.title.includes(filterText)
      );
      setFilteredProductList(filteredProducts);
    }
  }, [filterText, products]);

  useEffect(() => {
    shuffleArray(data);
  }, [data]);

  function shuffleArray(array) {
    for (let i = array.length - 1; i > 0; i--) {
      let j = Math.floor(Math.random() * (i + 1));
      let temp = array[i];
      array[i] = array[j];
      array[j] = temp;
    }
    return array;
  }

  return (
    <div className="container">
      <Header />
      {isLoading ? 'Loading' : ''}
      <Main
        products={filteredProductList}
        itemsInCart={props.itemsInCart}
        addToCart={props.addToCart}
        removeFromCart={props.removeFromCart}
        setFilter={setFilterText}
        filterText={filterText}
        submitCart={props.submitCart}
      />
      <Footer />
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    itemsInCart: state.cart.items,
    products: state.products.products,
  };
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators(actionCreators, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
