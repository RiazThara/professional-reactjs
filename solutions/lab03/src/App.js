import Footer from './Footer';
import BigOlHeader from './Header';
import Main from './Main';
import ListItems from './ListItems';

import './App.css';

function App() {
  return (
    <div className="App">
      <BigOlHeader />
      <ul>
        <ListItems />
      </ul>
      <Main />
      <Footer />
    </div>
  );
}

export default App;
