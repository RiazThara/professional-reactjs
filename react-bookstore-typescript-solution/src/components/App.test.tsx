import { render } from '@testing-library/react';
import App from '../components/App';

test('demo', () => {
  expect(true).toBe(true);
});

test('Renders the main page', () => {
  render(<App />);
  expect(true).toBeTruthy();
});

test('renders as expected', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});
