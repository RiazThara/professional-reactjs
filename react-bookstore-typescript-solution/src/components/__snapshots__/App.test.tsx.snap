// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      Welcome to React Bookstore.
    </h1>
    <p>
      Loading...
    </p>
    <div
      class="row"
    >
      <div
        class="col-md-8"
      >
        <ul
          class="productList"
        />
      </div>
      <div
        class="col-md-4"
      >
        <div
          class="cartStyle"
        >
          <h2>
            Cart
          </h2>
          Total: $
          0
        </div>
      </div>
    </div>
    <p
      class="text-color"
      style="background-color: black; color: white; padding: 10px; bottom: 0px; left: 0px; position: fixed; width: 100%;"
    >
      This is the footer.
    </p>
  </div>
</div>
`;
